src/user/user.controller.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { BadRequestException, NotFoundException } from "@nestjs/common";
+import { Test, TestingModule } from "@nestjs/testing";
+import { UserController } from "./user.controller";
+import { User } from "./user.entity";
+import { UserService } from "./user.service";
+
+class NoErrorThrownError extends Error {}
+
+async function getError<TError>(call: () => unknown): Promise<TError> {
```

Comment:
I can see us using this in other tests, I wonder if there is somewhere we could move it for reusability. Not sure where that would be though given tests live in `src`.